app/src/main/java/com/simon/harmonichackernews/CommentsFragment.java

```diff
@@ -464,6 +475,15 @@ public boolean onPreDraw() {
         });
     }
 
+    private void syncOnBackPressedCallbackEnabledState() {
```

Comment:
The logic determining whether the back handler is enabled was difficult to manage with this new option, so I introduced a single place to _sync_ the handler whenever any relevant state changes:
 - on webview page load
 - the bottom sheet state changes
 - in webview's `onPageFinished`